packages/hooks/src/useRefocusOnActivator/useRefocusOnActivator.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+
+/**
+ * Brings back the focus to the element that opened an overlaid element once
+ * said overlaid element is dismissed.
+ *
+ * @param active - Determines if it should focus or not
+ */
+export function useRefocusOnActivator(active: boolean) {
+  useEffect(() => {
+    let activator: Element | null | undefined;
+
+    if (active && !activator) {
+      activator = document.activeElement;
+    }
+
+    return () => {
+      if (active) {
+        if (activator instanceof HTMLElement) {
```

Comment:
What other types could `activator` be here?

Also, just a minor/nitpicky thing:  could two nested ifs make an `and` here maybe? 🤔

Reply:
hmm.. kinda torn between that tho. The `document.activeElement` is of an `Element` type which doesn't have the focus function. Instead of casting it as `as HTMLElement` I went with the safer way of making sure it is actually an `HTMLElement`. 

But yeah if it does return something that is not an `HTMLElement` the activator won't get wiped. To answer your question of what other types are out there, i don't know, hence it's separate haha.

Reply:
Okay!  That makes sense I think!